### PR TITLE
Results: Update income-related prompts and calculations for percentage of housing and transport costs

### PR DESCRIPTION
**Pull request by samuel-duhaime**
# Pull request

- [x] Update the tests with this.

## Description
Results: Update income-related prompts and calculations for percentage of housing and transport costs
Fix: #5, fix: #18
- Enhanced English and French prompts in home.yaml to clarify the use of household income information.
- Implemented a new function to calculate the percentage of income spent on housing and trasnport based on user input.
- Adjusted income range values in choices.tsx to reflect updated financial brackets.
- Updated results template to conditionally display percentage of income when available.
- Get only the minimum income for the high income household.

## Summary by CodeRabbit

* **New Features**
  * Percentage now reports combined housing + transport share of income (replaces housing-only metric)

* **Improvements**
  * Added explanatory clarification to the household income question (EN/FR)
  * Adjusted two household income range options and their labels
  * Wrapped percentage display in results with parentheses for clearer formatting
  * Results now include monthly housing and transport breakdowns and a total monthly cost

* **Tests**
  * Expanded unit and server tests covering varied income formats and edge cases

### CODE REVIEW

File: localisation/src/survey/common/choices.tsx [hunk 1]
```diff
@@ -194,17 +194,17 @@ export const householdIncomeChoices: ChoiceType[] = [
...
     },
     {
-        value: '200000_999999',
+        value: '210000_999999',
         label: {
-            fr: '200 000 $ et plus',
-            en: '$200,000 and more'
+            fr: '210 000 $ et plus',
+            en: '$210,000 and more'
         }
```
[the remark this answers is not in the document]
Comment (samuel-duhaime) [reply localisation/src/survey/common/choices.tsx]: Oh yeah good catch.

File: localisation/src/survey/calculations/index.ts [hunk 2]
```diff
@@ -68,13 +68,52 @@ const calculateMonthlyHousingCost = (address: Address): number | null => {
...
+
+    // Return null if the income range values are not numbers
+    if (isNaN(minIncome) || isNaN(maxIncome)) {
+        console.error('Invalid income range values:', income);
+        return null;
+    }
+
+    // Calculate the average of the income range
+    const averageAnnualIncome = (minIncome + maxIncome) / 2;
+
```
[the remark this answers is not in the document]
Comment (tahini) [reply localisation/src/survey/calculations/index.ts]: @samuel-duhaime il a un point pour le dernier bracket

File: localisation/src/survey/server/__tests__/serverFieldUpdate.test.ts [hunk 3]
```diff
@@ -666,4 +666,188 @@ describe('serverFieldUpdate - _sections._actions callback', () => {
             expect(mockCalculateAccessibilityAndRouting).toHaveBeenCalledWith(address2, interview);
         });
     });
+
+    describe('housing and transport cost percentage calculation', () => {
```
Comment (tahini) [inline localisation/src/survey/server/__tests__/serverFieldUpdate.test.ts]: I would be more important to add tests for the index.ts file, for the calculation itself, as indeed, these, in the `serverfieldUpdate` should just mock the results and validate the calls are done with the correct data. Anyway, you don't necessarily need to mock here are those cost calculations are rather simple. You can just let the code run.

Also, these tests look like a lot of copy paste, at least the test data should be defined outside the `it` part, in the describe, and ideally, use a parameterized test (but you could keep that for the `index.ts` tests).

**Review by tahini — COMMENTED** [review]
See the AI's comments. Also, the calculation functions should be well tested

**Review by tahini — APPROVED** [review]
